asyncio_mqtt/client.py

```diff
@@ -404,15 +406,19 @@ async def connect(self, *, timeout: int = 10) -> None:
         # See: https://github.com/eclipse/paho.mqtt.python/blob/v1.5.0/src/paho/mqtt/client.py#L1770
         except (OSError, mqtt.WebsocketConnectionError) as error:
             raise MqttError(str(error)) from None
-        await self._wait_for(self._connected, timeout=timeout)
+        await self._wait_for(
+            self._connected, timeout=self.timeout if timeout is None else timeout
```

Comment:
Again, good job with the "layered" timeout (use argument if given; otherwise, use instance default). Since we use that everywhere, let's move it inside `_wait_for` itself (to remove redundancy).

Inside `_wait_for` we have:

```python
if timeout is None:
    timeout = self._timeout
# Note that asyncio uses `None` to mean "No timeout". We use `math.inf`.
timeout_for_asyncio = None if timeout is math.inf else timeout
try:
    return await asyncio.wait_for(fut, timeout=timeout_for_asyncio, **kwargs)
except ...
```